packages/notebook/style/base.css

```diff
@@ -294,6 +294,65 @@
   flex: 0 0 110px;
 }
 
+/*-----------------------------------------------------------------------------
+| Side-by-side Mode (.jp-mod-sideBySide)
+|----------------------------------------------------------------------------*/
+:root {
+  --jp-side-by-side-output-size: 1fr;
+}
+
+.jp-mod-sideBySide.jp-Notebook .jp-CodeCell {
+  display: grid;
+  column-gap: var(--jp-cell-padding);
+  /* in side-by-side mode there are three rows: header, cell (input and output), footer */
+  grid-template-columns: minmax(0, 1fr) minmax(
+      0,
+      var(--jp-side-by-side-output-size)
+    );
+  /* in side-by-side mode there are two columns: input and output; header and footer span both of the columns */
+  grid-template-rows: auto minmax(0, 1fr) auto;
```

Comment:
note to self: comments for rows and columns are swapped in order; also maybe not needed since `grid-template-areas` below explains it quite well.